third_party/dependency_analyzer/src/test/io/bazel/rulesscala/dependencyanalyzer/AstUsedJarFinderTest.scala

```diff
@@ -302,6 +320,43 @@ class AstUsedJarFinderTest extends FunSuite {
     )
   }
 
+  test("inlined literal is direct") {
+    // Note: For a constant to be inlined
+    // - it must not have a type declaration such as `: Int`.
```

Comment:
The reason you're not adding a test for the case where there is a type declaration or a lower case capital letter because you're trying to take the strictest case, right?
If other cases are inlined then this checks inlining of literals already and if they won't be inlined in the future then it's a different feature?

Reply:
Yes (except not sure what you mean by "then it's a different feature")